CMakeLists.txt

```diff
@@ -1,198 +1,247 @@
-    include
-    ${OGRE_OV_INCLUDE_DIRS}
-)
-
-include_directories(SYSTEM include ${catkin_INCLUDE_DIRS} ${EIGEN3_INCLUDE_DIRS} ${OGRE_OV_INCLUDE_DIRS})
+find_package(Qt5 REQUIRED COMPONENTS Widgets)
 
 ###########
 ## Build ##
 ###########
-
-set(SOURCE_FILES
-  src/${PROJECT_NAME}_gui.cpp
-  src/key_tool.cpp
-)
+include_directories(
+  include
+  ${Eigen3_INCLUDE_DIRS})
 
 # Rviz GUI library
-add_library(${PROJECT_NAME}_gui ${SOURCE_FILES} src/remote_control.cpp)
-target_link_libraries(${PROJECT_NAME}_gui ${rviz_DEFAULT_PLUGIN_LIBRARIES} ${QT_LIBRARIES} ${catkin_LIBRARIES})
+# set(SOURCE_FILES
+  # src/${PROJECT_NAME}_gui.cpp
+  # src/key_tool.cpp
+# )
+# add_library(${PROJECT_NAME}_gui ${SOURCE_FILES} src/remote_control.cpp)
+# target_include_directories(${PROJECT_NAME}_gui PUBLIC
+  # ${OGRE_INCLUDE_DIRS}
+# )
+# target_link_libraries(${PROJECT_NAME}_gui Qt5::Widgets)
+# ament_target_dependencies(${PROJECT_NAME}_gui
+  # rclcpp
+# )
 
 # Remote control library
 add_library(${PROJECT_NAME}_remote_control src/remote_control.cpp)
-target_link_libraries(${PROJECT_NAME}_remote_control ${catkin_LIBRARIES})
+target_link_libraries(${PROJECT_NAME}_remote_control)
+ament_target_dependencies(${PROJECT_NAME}_remote_control
+  rclcpp
+)
 
 # Visualization Tools Library
-add_library(${PROJECT_NAME}
-  src/${PROJECT_NAME}.cpp
+add_library(${PROJECT_NAME} SHARED
+  src/rviz_visual_tools.cpp
+  src/remote_control.cpp
   src/tf_visual_tools.cpp
 )
+
 target_link_libraries(${PROJECT_NAME}
-  ${PROJECT_NAME}_remote_control
-  ${catkin_LIBRARIES}
   ${Boost_LIBRARIES}
 )
+ament_target_dependencies(${PROJECT_NAME}
+  rclcpp
+  rclcpp_components
+  visualization_msgs
+  tf2
+  tf2_eigen
+  tf2_geometry_msgs
+  sensor_msgs
+  std_msgs
+)
 
 # Library
 add_library(${PROJECT_NAME}_imarker_simple
   src/imarker_simple.cpp
+  src/rviz_visual_tools.cpp
+  src/remote_control.cpp
 )
 target_link_libraries(${PROJECT_NAME}_imarker_simple
-  ${catkin_LIBRARIES}
   ${Boost_LIBRARIES}
-  ${PROJECT_NAME}
+)
+ament_target_dependencies(${PROJECT_NAME}_imarker_simple
+  rclcpp
+  interactive_markers
+  geometry_msgs
+  visualization_msgs
+  sensor_msgs
 )
 
 # Demo executable
 add_executable(${PROJECT_NAME}_demo src/${PROJECT_NAME}_demo.cpp)
 target_link_libraries(${PROJECT_NAME}_demo
-  ${catkin_LIBRARIES} ${PROJECT_NAME}
+  ${PROJECT_NAME}
 )
-set_target_properties(${PROJECT_NAME}_demo
-  PROPERTIES
-  OUTPUT_NAME
-  demo PREFIX ""
+ament_target_dependencies(${PROJECT_NAME}_demo
+  rclcpp
+  geometry_msgs
+  std_msgs
 )
 
 # Demo executable
 add_executable(${PROJECT_NAME}_imarker_simple_demo src/imarker_simple_demo.cpp)
 target_link_libraries(${PROJECT_NAME}_imarker_simple_demo
-  ${catkin_LIBRARIES} ${PROJECT_NAME} ${PROJECT_NAME}_imarker_simple
-)
-set_target_properties(${PROJECT_NAME}_imarker_simple_demo
-  PROPERTIES
-  OUTPUT_NAME
-  imarker_simple_demo PREFIX ""
+  ${PROJECT_NAME}_imarker_simple
 )
 
-##########
-## TEST ##
-##########
-
-if(CATKIN_ENABLE_TESTING)
```

Comment:
Please do NOT disable the tests I worked so hard on! Reducing test coverage is not ok

Reply:
This is still a WIP as it creeps towards being feature complete. Wouldn't dream of merging until the tests were fixed ;)

Reply:
![giphy](https://user-images.githubusercontent.com/561060/72902435-72df9500-3cf9-11ea-818d-baf20c736c2d.gif)

Reply:
Really interestingly, ROS2 provides a bunch of CMake methods that do things like run clang format and clang tidy for you. This would actually make obsolete a lot of the features in moveit_ci. I took a quick pass at using ament_clang_format and got it to work correctly but it seems to act slightly differently than the clang format we use for moveit. Probably just a different release. Unfortunately, the two clang format approaches are not compatible so I kept the moveit_ci one for now. In the future we'll probably want to convert everything ros2 to use the ament methods